src/target/riscv/riscv-013.c

```diff
@@ -1726,6 +1759,12 @@ static int examine(struct target *target)
 		dmi_write(target, DM_DMCONTROL, DM_DMCONTROL_DMACTIVE);
 		dm->was_reset = true;
 	}
+
+	if (!check_dbgbase_exists(target)) {
```

Comment:
This should happen before we attempt any other accesses to the DM.

Reply:
done